File: Angular/src/app/Model/ProductModel.ts
export class ProductModel{
    id: number = 0;
    Mode:string="";
    image !: any;
    Name: string = "";
    Category: string = "";
    Description: string = "";
    Quantity: number = 0;
    Manufacturing_Date: Date = new Date();
    IsOutofstock: boolean = false;
    Color: string = "";
    Price: number = 0;
    IsActive:boolean=true;
    CreatedBy: string = '';
    CreatedDate: Date = new Date();
    ModifiedBy: string = '';
    ModifiedDate: Date = new Date();
}

File: Api/Demo_Product.Model/Models/ProductModel.cs
using System.ComponentModel.DataAnnotations;

namespace Demo_Product.Model.Models
{
    public class ProductModel
    {
        [Key]
        public int id { get; set; }
        public string? Image { get; set; }

        public string Name { get; set; }

        public string Category { get; set; }
        public string? CategoryName { get; set; }

        public string Description { get; set; }

        public int Quantity { get; set; }

        public DateTime Manufacturing_Date { get; set; }

        public bool IsOutofstock { get; set; }

        public string Color { get; set; }

        public double Price { get; set; }
        public string Mode { get; set; }
        public bool? IsActive { get; set; }
        public DateTime? CreatedDate { get; set; }
        public string CreatedBy { get; set; }
        public DateTime? ModifiedDate { get; set; }
        public string? ModifiedBy { get; set; }

    }
}
